From afd06da37bd6db14fd01263a89a22a1e913e38d1 Mon Sep 17 00:00:00 2001
From: Luciana Fujii Pontello <lucianafujii@collabora.co.uk>
Date: Mon, 7 Nov 2011 21:34:43 -0500
Subject: [PATCH 120/154] Don't fail to build if timemmgr is not present

Check for timemmgr only to build pvrvideosink, if it's not installed
pvrvideosink is not built.
---
 configure.ac    |   13 +++++++++----
 sys/Makefile.am |    4 ++++
 2 files changed, 13 insertions(+), 4 deletions(-)

diff --git a/configure.ac b/configure.ac
index 6ce51f3..7d9ad94 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1306,10 +1306,14 @@ AG_GST_CHECK_FEATURE(OPENCV, [opencv plugins], opencv, [
   AC_SUBST(OPENCV_LIBS)
 ])
 
-dnl *** PVR ***
- PKG_CHECK_MODULES([PVR], [libtimemmgr])
- AC_SUBST(PVR_CFLAGS)
- AC_SUBST(PVR_LIBS)
+dnl *** pvr ***
+translit(dnm, m, l) AM_CONDITIONAL(USE_PVR, true)
+AG_GST_CHECK_FEATURE(PVR, [pvrvideosink], pvr, [
+  PKG_CHECK_MODULES([PVR], [libtimemmgr], HAVE_PVR=yes, HAVE_PVR=no)
+  AC_SUBST(PVR_CFLAGS)
+  AC_SUBST(PVR_LIBS)
+])
+
  AC_ARG_WITH([pvr-external-headers],
               AC_HELP_STRING([--with-pvr-external-headers], [Use system installed PVR2D headers]),
               [AS_IF([test "x$with_pvr_external_headers" = "xno"],
@@ -1671,6 +1675,7 @@ AM_CONDITIONAL(USE_NAS, false)
 AM_CONDITIONAL(USE_NEON, false)
 AM_CONDITIONAL(USE_OFA, false)
 AM_CONDITIONAL(USE_OPENCV, false)
+AM_CONDITIONAL(USE_PVR, false)
 AM_CONDITIONAL(USE_RSVG, false)
 AM_CONDITIONAL(USE_TIMIDITY, false)
 AM_CONDITIONAL(USE_WILDMIDI, false)
diff --git a/sys/Makefile.am b/sys/Makefile.am
index c1ffad7..d6f678f 100644
--- a/sys/Makefile.am
+++ b/sys/Makefile.am
@@ -102,7 +102,11 @@ else
 SHM_DIR=
 endif
 
+if USE_PVR
 PVR_DIR=pvr2d
+else
+PVR_DIR=
+endif
 
 SUBDIRS = $(ACM_DIR) $(APPLE_MEDIA_DIR) $(DECKLINK_DIR) $(DIRECTDRAW_DIR) $(DIRECTSOUND_DIR) $(DVB_DIR) $(FBDEV_DIR) $(LINSYS_DIR) $(OSX_VIDEO_DIR) $(QT_DIR) $(SHM_DIR) $(VCD_DIR) $(VDPAU_DIR) $(WININET_DIR) $(PVR_DIR)
 
-- 
1.7.5.4

